src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,21 @@
+import PopupWithForm from "./PopupWithForm.js";
+
+export default class PopupWithConfirmation extends PopupWithForm {
+  constructor(popupSelector, handleFormSubmit, buttonText, loadingButtonText) {
+    super(popupSelector, handleFormSubmit, buttonText, loadingButtonText);
+  }
+
+  open(data) {
+    super.open();
+    this.data = data;
+  }
+
+  setSubmitAction(handleFormSubmit) {
+    this._handleFormSubmit = handleFormSubmit;
+  }
+
+  _handleSubmit = () => {
+    const inputValues = this.data;
+    this._handleFormSubmit(inputValues);
+  };
+}
```

Comment:
You should also implement loading functionality here, like in PopupWithForm class

Reply:
Isn't showLoading() and hideLoading() inherited from the PopupWithForm class? Do I need to add these methods?